cloud/google/cmd/gce-controller/main.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"github.com/golang/glog"
+	"github.com/spf13/pflag"
+	"k8s.io/apiserver/pkg/util/logs"
+	"sigs.k8s.io/cluster-api/cloud/google/cmd/gce-controller/cluster-controller-app"
+	clusteroptions "sigs.k8s.io/cluster-api/cloud/google/cmd/gce-controller/cluster-controller-app/options"
+	"sigs.k8s.io/cluster-api/cloud/google/cmd/gce-controller/machine-controller-app"
+	machineoptions "sigs.k8s.io/cluster-api/cloud/google/cmd/gce-controller/machine-controller-app/options"
+	"sigs.k8s.io/cluster-api/pkg/controller/config"
+)
+
+func main() {
+
+	fs := pflag.CommandLine
+	var controllerType, kubeadmToken, machineSetupConfigsPath string
+	fs.StringVar(&controllerType, "controller", controllerType, "speficy whether this should run the machine or cluster controller")
+	fs.StringVar(&kubeadmToken, "token", kubeadmToken, "Kubeadm token to use to join new machines")
+	fs.StringVar(&machineSetupConfigsPath, "machinesetup", machineSetupConfigsPath, "path to machine setup configs file")
+	config.ControllerConfig.AddFlags(pflag.CommandLine)
+	pflag.Parse()
```

Comment:
@k4leung4 Does this look less confusing?

Reply:
this looks better, thanks